crates/subspace-farmer/src/commitments/commitment_databases.rs

```diff
@@ -40,10 +37,14 @@ impl Deref for DbEntry {
     }
 }
 
-#[derive(Debug)]
 pub(super) struct CommitmentDatabases {
     base_directory: PathBuf,
-    databases: LruCache<Salt, Arc<DbEntry>>,
+    first_db: RwLock<Option<Arc<DbEntry>>>,
+    second_db: RwLock<Option<Arc<DbEntry>>>,
+    metadata: RwLock<CommitmentMetadata>,
```

Comment:
Instead of 2 `RwLock`s, I put these together in a struct, and acquiring the necessary lock on this struct.